### PR TITLE
create PostConstruct method runScript() in DBConfig

### PR DESCRIPTION
**Pull request by evyrva**
create script for db in resources/db

### CODE REVIEW

File: src/main/java/com/epam/webelecty/config/DBConfig.java [hunk 1]
```diff
@@ -27,5 +36,47 @@ public ConnectionPool connectionPool() {
                 driverName, url, user, password, poolSize);
     }
 
+    @PostConstruct
+    public void runScript() {
+        String sqlBaseInit = getSQLFromFile("src/main/resources/db/dataBaseInitH2.sql");
+        String sqlFillDB = getSQLFromFile("src/main/resources/db/fillDBWithTestDataH2.sql");
```
Comment (SMishAlex) [inline src/main/java/com/epam/webelecty/config/DBConfig.java]: better move those literals to property file

File: pom.xml [hunk 2]
```diff
@@ -119,7 +119,7 @@
             <groupId>com.h2database</groupId>
             <artifactId>h2</artifactId>
             <version>1.4.196</version>
-            <scope>test</scope>
+            <!--<scope>test</scope>-->
```
Comment (SMishAlex) [inline pom.xml]: just remove if u don't need it

File: src/main/java/com/epam/webelecty/config/DBConfig.java [hunk 3]
```diff
@@ -27,5 +36,47 @@ public ConnectionPool connectionPool() {
...
+        DAO dao = new DAO() {
```
Comment (SMishAlex) [inline src/main/java/com/epam/webelecty/config/DBConfig.java]: it looks bad.. better make executeSqlStatement static. if it is hard dont return null throw unsupported operation exception instead

File: src/main/java/com/epam/webelecty/config/DBConfig.java [hunk 4]
```diff
@@ -27,5 +36,47 @@ public ConnectionPool connectionPool() {
...
+            }
+        };
+        dao.executeSqlStatement(connectionPool(), sqlBaseInit);
+        dao.executeSqlStatement(connectionPool(), sqlFillDB);
+    }
+
+    private String getSQLFromFile(String file) {
+        StringBuilder sql = new StringBuilder("");
+        try {
+            Files.lines(Paths.get(file)).forEach(sql::append);
```
Comment (SMishAlex) [inline src/main/java/com/epam/webelecty/config/DBConfig.java]: use .collect(Collectors.joining()) instead

Comment (SMishAlex) [inline src/main/java/com/epam/webelecty/config/DBConfig.java]: The returned stream encapsulates a {@link Reader}.  If timely
     * disposal of file system resources is required, the try-with-resources
     * construct should be used to ensure that the stream's
     * {@link Stream#close close} method is invoked after the stream operations
     * are completed.

this mean you should use try with resources here

**Review by SMishAlex — APPROVED** [review]

File: src/main/java/com/epam/webelecty/config/DBConfig.java [hunk 5]
```diff
@@ -20,12 +31,56 @@
...
     public ConnectionPool connectionPool() {
         return new ConnectionPool(
                 driverName, url, user, password, poolSize);
     }
 
+    @PostConstruct
+    public void runScript() {
+        DAO dao = new DAO() {
+            @Override
+            public Set getAllEntries() {
```
Comment (petriankins) [inline src/main/java/com/epam/webelecty/config/DBConfig.java]: what is this???

File: src/main/java/com/epam/webelecty/controllers/TestController.java [hunk 6]
```diff
@@ -64,7 +64,7 @@ public String registration(Model model) {
     @RequestMapping(value = "/registration", method = RequestMethod.POST)
     public String registration(@ModelAttribute("userForm") UserDTO userForm) {
         userService.register(userForm);
-        return "redirect:/index";
+        return "redirect:/user";
```
Comment (petriankins) [inline src/main/java/com/epam/webelecty/controllers/TestController.java]: why are you changing it in your branch? it's not about h2 at all

File: src/main/resources/db/dataBaseInitH2.sql [hunk 7]
```diff
@@ -0,0 +1,32 @@
+CREATE SCHEMA test;
+
+CREATE TABLE test.users (
+  userId   INT AUTO_INCREMENT,
+  email    VARCHAR(45)  NOT NULL UNIQUE,
+  pass     VARCHAR(244) NOT NULL,
+  name     VARCHAR(45),
+  lastName VARCHAR(45),
+  role     VARCHAR(45)  NOT NULL,
```
Comment (petriankins) [inline src/main/resources/db/dataBaseInitH2.sql]: do we need 45 symbols in our role?

File: src/main/resources/db/fillDBWithTestDataH2.sql [hunk 8]
```diff
@@ -0,0 +1,55 @@
+INSERT INTO test.users (email, pass, name, lastName, role)
+VALUES ('s1@mail.ru', '$2a$05$ExEIw8WuiAAP1ZZFb3FYx.nK6q7FOzX4I/XskYNNihOEv6qk.xy7S', 'Ivan', 'Ivanov', 'student'),
+  ('s2@mail.ru', '$2a$05$ExEIw8WuiAAP1ZZFb3FYx.nK6q7FOzX4I/XskYNNihOEv6qk.xy7S', 'Zefir', 'Marmeladov', 'student'),
+  ('s3@mail.ru', '$2a$05$ExEIw8WuiAAP1ZZFb3FYx.nK6q7FOzX4I/XskYNNihOEv6qk.xy7S', 'Venya', 'Erofeev', 'student'),
```
Comment (petriankins) [inline src/main/resources/db/fillDBWithTestDataH2.sql:4]: i nemedlenno vypil

**Pull request merged**